webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -78,7 +78,7 @@ export class MonacoService implements OnDestroy {
         e.preventDefault();
         let fileContext = self.editorControl.fetchActiveFile();
         let directory = fileContext.model.path || self.editorControl.activeDirectory;
-        let sub = self.saveFile(fileContext, directory).subscribe(() => sub.unsubscribe());
+        let sub = self.saveFile(fileContext, directory).subscribe(() => {}, e => {}, () => {}); // Error handling is done up-stream
```

Comment:
You have to either call `obs.complete()` after each `obs.next()` inside `saveFile()` or do `sub.unsibscribe()` otherwise it's a memory leak.

Reply:
Are you sure about this? I changed it because of this: https://angular.io/guide/observables

![image](https://user-images.githubusercontent.com/20528015/133842874-4438460a-1dea-418c-8ad0-360f5049361c.png)